### PR TITLE
Migration of work loader with changes for migration

### LINKED ISSUE #1300: Migrate batch load and edit code from 2.x
Migrate the code from Scholar 2.x that allows batch load and edit.

Not needed for the beta preview

### LINKED ISSUE #1596: Add collection loader
Model after work in feature/#1300-migrate-batch-loader

### LINKED ISSUE #1608: Extend WorkLoader class to handle work relationships

### PR DESCRIPTION
**Pull request by jamesvanmil**
Closes #1300 
Closes #1608 
Closes #1596

This will continue to get debugged in the near future, but let's start a review now?

### CODE REVIEW

**Comment by jamesvanmil** [comment]
@crowesn I had left the [WIP] off of this because I had asked for someone to start reviewing

File: lib/tasks/batch.rake [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          puts "Failed to create: #{wl.work_log}"
+        end
+        remove_tmp_files(work[:files])
+      end
+
+      close_csv work_log
+      close_csv file_log
+    end
+
+    # rake batch:load:collections CSV_LOCATION="/Users/vanmiljf/Downloads/sample_batch_metadata.txt" PATH_TO_AVATARS="/home/james/batch"
```
Comment (crowesn) [inline lib/tasks/batch.rake]: clean up our comments.

File: lib/tasks/batch.rake [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    desc 'Load multiple collections from a tab delimited text file'
+    task collections: :environment do
+      #collection_log = new_collection_upload_log
```
Comment (crowesn) [inline lib/tasks/batch.rake]: Enable collection logging.

File: lib/tasks/batch.rake [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      collections.each do |collection|
+        file_path = collection.delete(:avatar_path)
+        unless file_path.nil?
+          file_path = File.join(ENV["PATH_TO_AVATARS"], file_path)
+          rsync_to_tmp file_path
+          collection[:avatar_path] = tmp file_path
+        end
+        cl = CollectionLoader.new(collection.dup)
+        if cl.create
+          puts "Success"
```
Comment (crowesn) [inline lib/tasks/batch.rake]: Log more details.

**Pull request merged**